spec/service_worker/index.bs

```diff
@@ -1849,6 +1843,74 @@ spec: webidl; urlPrefix: https://heycam.github.io/webidl/
+    </ol>
+
+    <p>When a <a>serviceworker link</a>'s <{link}> element is <a lt="in a document">inserted into a document</a>, a <a>serviceworker link</a> is created on a <{link}> element that is already <a>in a Document</a>, or the <{link/href}> or <{link/scope}> attributes of the <{link}> element of a <a>serviceworker link</a> is changed, the user agent <em class="rfc2119" title="SHOULD">should</em> run these steps or their <a href="#dfn-processing-equivalence">equivalent</a>:</p>
+
+    <ol>
+      <li>If the <{link/href}> attribute is the empty string, abort these steps.</li>
+      <li>Let <var>client</var> be the document's <a for="ServiceWorkerContainer">service worker client</a>.</li>
+      <li>Let <var>scriptURL</var> be the result of <a for="url">parsing</a> the <{link/href}> attribute with document's <a>document base URL</a>.</li>
+      <li>Let <var>scopeURL</var> be the <{link/scope}> attribute, or null if the <{link/scope}> attribute is omitted.</li>
+      <li>Let <var>promise</var> be a new <a>promise</a>.</li>
+      <li>Invoke [[#start-register-algorithm]] with <var>scopeURL</var>, <var>scriptURL</var>, <var>promise</var>, <var>client</var>, and <var>client</var>'s <a for="resource">origin</a>.</li>
+      <li>Run the following substeps <a>in parallel</a>:
+        <ol>
+          <li>Wait until <var>promise</var> settles.</li>
+          <li>If <var>promise</var> rejected, <a>queue a task</a> to <a>fire a simple event</a> named <code>error</code> at the <{link}> element.</li>
+          <li>If <var>promise</var> resolved, <a>queue a task</a> to <a>fire a simple event</a> named <code>load</code> at the <{link}> element.</li>
+        </ol>
+      </li>
+    </ol>
+
+    <p>The <a href="#dfn-serviceworker-link">serviceworker link</a> element <em class="rfc2119" title="MUST NOT">must not</em> <a href="https://html.spec.whatwg.org/multipage/syntax.html#delay-the-load-event">delay the load event</a> of the element's <a href="https://dom.spec.whatwg.org/#concept-node-document">node document</a>.</p>
+
+    <div class="example">
+      A resource being loaded with the following response header:
+      <pre class="highlight">
+        Link: &lt;/js/sw.js&gt;; rel="serviceworker"; scope="/"
+      </pre>
+      has more or less the same effect as a document being loaded in a secure context with the following <code>link</code> element:
+      <pre class="lang-html">
+        &lt;link rel="serviceworker" href="/js/sw.js" scope="/"&gt;
+      </pre>
+      which is more or less equivalent to the page containing javascript code like:
+      <pre class="lang-js">
+        navigator.serviceworker.register("/js/sw.js", { scope: "/" });
+      </pre>
+    </div>
+  </section>
+
+  <section>
+    <h3 id="link-element-interface-section">Link element interface extensions</h3>
+
+    <pre class="idl">
+      partial interface HTMLLinkElement {
+        attribute USVString scope;
```

Comment:
Can you add [CEReactions] to this, per https://github.com/w3c/webcomponents/issues/186 ?